internal/tiltfile/tiltfile_test.go

```diff
@@ -3836,57 +3837,61 @@ k8s_yaml('foo.yaml')
 	f := newFixture(t)
 	defer f.TearDown()
 
 	f.gitInit("")
-	f.file("resource.yaml", fmt.Sprintf(`
-%s
----
-%s
-`, testyaml.DoggosServiceYaml, testyaml.DoggosServiceYaml))
+	f.yaml("resource.yaml",
+		service("doggos"),
+		service("doggos"))
 	f.file("Tiltfile", `
 k8s_yaml('resource.yaml')
 `)
-	f.loadAllowWarnings()
-	m := f.assertNextManifestUnresourced("doggos", "doggos")
-	displayNames := []string{}
-	displayNames = append(displayNames, m.K8sTarget().DisplayNames...)
-	assert.Equal(t, []string{"doggos:service:default:core:0", "doggos:service:default:core:1"}, displayNames)
-
-	duplicateWarningStr := "Resource uncategorized contains multiple specifications of k8s entity(s): doggos:service:default:core. Only one specification per entity can be applied to the cluster; to ensure expected behavior, remove the duplicate specifications"
-	f.assertWarnings(duplicateWarningStr)
+	f.loadErrString(k8s.DuplicateYamlDetectedError("doggos:service:default:core"))
 }
-
-func TestMultipleDuplicateResources(t *testing.T) {
+func TestDuplicateYamlEntityAcrossResources(t *testing.T) {
 	f := newFixture(t)
 	defer f.TearDown()
 
-	f.gitInit("")
-	f.file("resource.yaml", fmt.Sprintf(`
-%s
----
-%s
----
-%s
----
-%s
-`, testyaml.DoggosServiceYaml, testyaml.DoggosServiceYaml, testyaml.CatsServiceYaml, testyaml.CatsServiceYaml))
+	f.dockerfile("foo1/Dockerfile")
+	f.yaml("foo1.yaml", deployment("foo", image("gcr.io/foo1"), namespace("ns1")))
 	f.file("Tiltfile", `
-k8s_yaml('resource.yaml')
-`)
-	f.loadAllowWarnings()
 
-	m := f.assertNextManifestUnresourced("doggos", "doggos", "cats", "cats")
-	displayNames := []string{}
-	displayNames = append(displayNames, m.K8sTarget().DisplayNames...)
-	assert.Equal(t, []string{"doggos:service:default:core:0", "doggos:service:default:core:1", "cats:service:default:core:2", "cats:service:default:core:3"}, displayNames)
+k8s_yaml(['foo1.yaml', 'foo1.yaml'])
+k8s_resource('foo:deployment:ns1', new_name='foo')
+`)
+	f.loadErrString(k8s.DuplicateYamlDetectedError("foo:deployment:ns1:apps"))
+}
 
-	duplicateWarningStr := "Resource uncategorized contains multiple specifications of k8s entity(s): doggos:service:default:core, cats:service:default:core. Only one specification per entity can be applied to the cluster; to ensure expected behavior, remove the duplicate specifications"
+func TestDuplicateYamlEntityInSingleWorkload(t *testing.T) {
+	//Services corresponding to a deployment get pulled into the same resource.
+	f := newFixture(t)
+	defer f.TearDown()
 
-	f.assertWarnings(duplicateWarningStr)
+	labelsFoo := map[string]string{"foo": "bar"}
+	f.yaml("all.yaml",
+		deployment("foo-deployment", image("gcr.io/foo1"), namespace("ns1"), withLabels(labelsFoo)),
+		service("foo-service", withLabels(labelsFoo)),
+		service("foo-service", withLabels(labelsFoo)))
+	f.file("Tiltfile", `
+k8s_yaml('all.yaml')
+`)
+	f.loadErrString(k8s.DuplicateYamlDetectedError("foo-service:service:default:core"))
```

Comment:
@jazzdan This seemed to be a good way to get access to that const....should I re-declare it in this file, or is the current method okay?

Reply:
Ah! I missed this. Yeah, makes sense! My question was a genuine one, not a gotcha :sweat_smile: